list.go

```diff
@@ -27,14 +27,58 @@ import (
 // List
 /////////////////////////////////////////
 
-// # list/vector
+// encList write list
+func (e *Encoder) encList(v interface{}) error {
+	if reflect.TypeOf(v).String() != "[]interface {}" {
+		if e.writeTypedList(v) == nil {
```

Comment:
should return error when not nil

Reply:
When some types are not supported for encoding to typedList， it will try to encode to untypedList, so I ignore it's error.

Reply:
@fangyincheng I agree with wongoo's opinion. Further more, if encoding typed list fails, and encoder will cosume byte, then encoding untyped list may raise EOF error directly.

Reply:
@wongoo @gaoxinge 
I will modify it as follows: 
>noType = perrors.New("no type name.")
> if err := e.writeTypedList(v); err != noType {
			return err
		}
What do you think?

Reply:
@fangyincheng should it do automatically check and encode as untyped list?  I don't think so, instead, can it do auto register for types?

Reply:
You are right， but I don‘t think it can auto register for types. I will return error. OK?

Reply:
@fangyincheng  agree